lib/mongoidable/document_extensions.rb

```diff
@@ -6,10 +6,71 @@ module DocumentExtensions
     extend ActiveSupport::Concern
 
     included do
-      ability_class = Mongoidable.configuration.ability_class.constantize
+      ability_class = Mongoidable.configuration.ability_class
       raise TypeError unless ability_class == Mongoidable::Ability || ability_class.superclass == Mongoidable::Ability
 
-      embeds_many :instance_abilities, class_name: Mongoidable.configuration.ability_class
+      embeds_many :instance_abilities, class_name: Mongoidable.configuration.ability_class.to_s do
+        def update(**attributes)
+          return unless should_update?(attributes)
+
+          attributes = ActionController::Parameters.new(attributes) if attributes.is_a?(Hash)
+          query = ability_query_attrs(attributes)
+
+          #
+          ability = instance_abilities.where(query).first
+          type = ability_type(attributes[:action])
+          method = parent_document.new_record? ? :build : :create
+
+
+          create_attributes =  ability_create_attrs(attributes)
+          ability = instance_abilities.where(query).first || instance_abilities.send(method, create_attributes, type)
+
+          update_attrs = ability_update_attrs(ability, attributes)
+          ability.update_attributes update_attrs
```

Comment:
Or maybe did you intend this to be an `assign_attributes` if the `parent_document.new_record?`?